pkgs/tools/archivers/7zz/default.nix

```diff
@@ -1,23 +1,30 @@
 { stdenv, lib, fetchurl, p7zip }:
 
+let
+  # we need https://github.com/nidud/asmc/tree/master/source/asmc/linux in order
+  # to build with the optimized assembler but that doesn't support building with
+  # GCC: https://github.com/nidud/asmc/issues/8
+  makefile = {
+    "aarch64-darwin" = "../../cmpl_mac_arm64.mak";
+    "x86_64-darwin" = "../../cmpl_mac_x64.mak";
+  }.${stdenv.hostPlatform.system} or "../../cmpl_gcc.mak";
+in
 # https://sourceforge.net/p/sevenzip/discussion/45797/thread/7fe6c21efa/
 stdenv.mkDerivation rec {
   pname = "7zz";
   version = "21.04";
 
   src = fetchurl {
-    url = "https://7-zip.org/a/7z${lib.replaceStrings ["." ] [""] version}-src.7z";
+    url = "https://7-zip.org/a/7z${lib.replaceStrings [ "." ] [ "" ] version}-src.7z";
     sha256 = "sha256-XmuEyIJAJQM0ZbgrW02lQ2rp4KFDBjLXKRaTfY+VCOg=";
   };
 
   sourceRoot = "CPP/7zip/Bundles/Alone2";
 
-  # we need https://github.com/nidud/asmc/tree/master/source/asmc/linux in order
-  # to build with the optimized assembler but that doesn't support building with
-  # GCC: https://github.com/nidud/asmc/issues/8
-  makefile = "../../cmpl_gcc.mak"; # "../../cmpl_gcc_x64.mak";
+  inherit makefile;
 
-  NIX_CFLAGS_COMPILE = [ "-Wno-error=maybe-uninitialized" ];
+  NIX_CFLAGS_COMPILE = [ "-Wno-error=maybe-uninitialized" ] ++
+    lib.optional stdenv.cc.isClang "-Wno-unknown-warning-option";
```

Comment:
Without this we have multiple errors for unknown options in `clang`. However, I think this only happens because our Clang is old.

Maybe once [this PR is merged](https://github.com/NixOS/nixpkgs/pull/126411), we can remove this option.

Reply:
Please put this comment in

Reply:
The idea isn't to merge this PR as-is because it doesn't build. This is why this is set to draft.

Reply:
But sure, done.